engine/src/org/pentaho/di/trans/steps/excelwriter/ExcelWriterStep.java

```diff
@@ -89,8 +90,13 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
     if ( first ) {
 
       first = false;
-      data.outputRowMeta = getInputRowMeta().clone();
-      data.inputRowMeta = getInputRowMeta().clone();
+      if ( r == null ) {
```

Comment:
Don't think this belongs here. If you look at other steps, the pattern is more like:

Object[] r = getRow();
if  ( r == null ) {
  // no more input to be expected
  setOutputDone();
  return false;
}
...
if ( first ) {